syft/pkg/cataloger/debian/parse_deb_archive.go

```diff
@@ -0,0 +1,268 @@
+package debian
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"path/filepath"
+	"strings"
+
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/generic"
+	"github.com/blakesmith/ar"
+	"github.com/klauspost/compress/zstd"
+	"github.com/ulikunitz/xz"
+)
+
+// parseDebArchive parses a Debian package archive (.deb) file and returns the packages it contains.
+// A .deb file is an ar archive containing three main files:
+// - debian-binary: Version of the .deb format (usually "2.0")
+// - control.tar.gz/xz/zst: Contains package metadata (control file, md5sums, conffiles)
+// - data.tar.gz/xz/zst: Contains the actual files to be installed (not processed by this cataloger)
+//
+// This function extracts and processes the control information to create package metadata.
+func parseDebArchive(_ context.Context, _ file.Resolver, _ *generic.Environment, reader file.LocationReadCloser) ([]pkg.Package, []artifact.Relationship, error) {
```

Comment:
this should be decomposed into smaller functions to help with readability (linting is primarily failing for that reason)